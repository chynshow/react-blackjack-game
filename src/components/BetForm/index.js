import React, { useContext, useState } from 'react';
import { AppContext } from '../../state/AppContext';
import { SHOW_INFO_MODAL } from '../../state/AppReducer';

const BetForm = () => {
  const { setBet, dispatch, playerCredit, gameStarted } = useContext(
    AppContext
  );
  const [value, setValue] = useState(0);

  const handleOnSubmit = (e) => {
    e.preventDefault();
    if (gameStarted && (!value || value > playerCredit))
      return dispatch({
        type: SHOW_INFO_MODAL,
        payload: {
          title: 'Please set a correct bet!',
          msg: `Bet should be more then zero and less then ${playerCredit}$`,
          closeBtnTitle: 'Back to game!',
        },
      });
    setBet(value);
  };
  return (
    <form className='c-bet-form' onSubmit={(e) => handleOnSubmit(e)}>
      <input
        className='c-bet-form__input'
        value={value}
        type='number'
        onChange={(e) => setValue(+e.target.value)}
      />
      <button type='submit' className='c-btn c-bet-form__btn'>
        Set Bet $
      </button>
    </form>
  );
};

export default BetForm;
