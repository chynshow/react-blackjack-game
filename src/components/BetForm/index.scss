.c-bet-form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1rem 0;

  &__input {
    margin: 1rem 0;
    border: 0;
    outline: 0;
    width: 6rem;
    height: 6rem;
    border-radius: 50%;
    text-align: center;
    cursor: pointer;
    color: $text-color;
    font-size: 1.5rem;
    background: transparent;
    box-shadow: inset 4px 4px 10px rgba($color: #000000, $alpha: 0.1),
      inset 8px 8px 10px rgba($color: #000000, $alpha: 0.1);

    &::-webkit-outer-spin-button,
    &::-webkit-inner-spin-button {
      -webkit-appearance: none;
      margin: 0;
    }

    &[type='number'] {
      -moz-appearance: textfield;
    }
    &:disabled {
      opacity: 0.5;
      pointer-events: none;
    }
  }

  &__btn {
    margin: 0.5rem 0;
  }
}
