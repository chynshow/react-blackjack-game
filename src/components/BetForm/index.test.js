import React from 'react';
import BetForm from './index';
import { shallow } from 'enzyme';

it('render BetForm component', () => {
  const c = shallow(<BetForm />);
  expect(c).toMatchSnapshot();
});
