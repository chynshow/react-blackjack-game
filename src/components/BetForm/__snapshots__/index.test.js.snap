// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render BetForm component 1`] = `
<form
  className="c-bet-form"
  onSubmit={[Function]}
>
  <input
    className="c-bet-form__input"
    onChange={[Function]}
    type="number"
    value={0}
  />
  <button
    className="c-btn c-bet-form__btn"
    type="submit"
  >
    Set Bet $
  </button>
</form>
`;
